### PR TITLE
Perform reverse lookup of origin IPs on log retrieval

### PR DESCRIPTION
**Pull request by cpaillet**
OriginIP is an IP, it's not very human friendly for dashboard. We can resolve this ip and give the reverse dns name if available.

### CODE REVIEW

**Review by dclaisse — COMMENTED** [review]
I don't see the point of extracting hostnames in the exporter and add complexity. Could you please elaborate on what problem are you trying to fix?

**Comment by cpaillet** [comment]
first comment updated

File: cloudflare_exporter/collector.py [hunk 1]
```diff
@@ -167,10 +171,30 @@ def __init__(self):
         self.received_requests_pop_origin = Counter()
         self.origin_response_times = defaultdict(list)
 
+    @staticmethod
+    @lru_cache(maxsize=1000)
```
Comment (pierrecdn) [inline cloudflare_exporter/collector.py]: Why is this cache needed? (especially if cleaned at each run)

**Review by pierrecdn — COMMENTED** [review]
Remark: We do not create PTR entries for now, so this will mostly not give results.
Also, we could use our internal tooling to fetch naming of VIPs instead of depending on a system we provision, but this would be out of the scope of this public repo.

File: cloudflare_exporter/collector.py (hunk 1 above)
Comment (cpaillet) [reply cloudflare_exporter/collector.py]: because we call this function on each log line, and I don't want to make a dns call for each line !

**Comment by cpaillet** [comment]
@pierrecdn I already create PTR for the current cloudflare VIP.

File: cloudflare_exporter/collector.py (hunk 1 above)
Comment (pierrecdn) [reply cloudflare_exporter/collector.py]: OK, please extract the maxsize into a constant or whatever that can ease this change for the future maintainer.

**Review by pierrecdn — CHANGES_REQUESTED** [review]
This remark, also could you please reword your PR? 
"Get dns reverse of OriginIP" > "Perform reverse lookup of origin IPs on log retrieval", "Retrieve hostname by performing reverse lookups on log lines"... And use the body field to explicit.
Your choice for the wording, but more explicit.
I know what you're doing because I have context, someone else browsing the repo should too.

**Comment by dclaisse** [comment]
Still not seeing why you need to do reverse DNS lookups, could you please explain it?

**Comment by cpaillet** [comment]
it's on the PR comment -> OriginIP is an IP, it's not very human friendly for dashboard. We can resolve this ip and give the reverse dns name if available.

**Review by pierrecdn — APPROVED** [review]

**Pull request merged**